src/plugins/vis_default_editor/public/components/controls/has_extended_bounds.tsx

```diff
@@ -17,22 +17,32 @@
  * under the License.
  */
 
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 import { i18n } from '@kbn/i18n';
 
-import { search } from '../../../../../../plugins/data/public';
+import { search } from '../../../../data/public';
 import { SwitchParamEditor } from './switch';
 import { AggParamEditorProps } from '../agg_param_props';
 
 const { isType } = search.aggs;
 
 function HasExtendedBoundsParamEditor(props: AggParamEditorProps<boolean>) {
+  const { agg, setValue, value } = props;
+  const minDocCount = useRef(agg.params.min_doc_count);
+
   useEffect(() => {
-    props.setValue(props.value && props.agg.params.min_doc_count);
-  }, [props.agg.params.min_doc_count]);
+    if (minDocCount.current !== agg.params.min_doc_count) {
+      // The "Extend bounds" param is only enabled when "Show empty buckets" is turned on.
+      // So if "Show empty buckets" is changed, "Extend bounds" should reflect changes
+      minDocCount.current = agg.params.min_doc_count;
+
+      setValue(value && agg.params.min_doc_count);
+    }
+  }, [agg.params.min_doc_count, setValue, value]);
 
   return (
     <SwitchParamEditor
+      {...props}
```

Comment:
This was moved upper since `props` has a `disabled` field which overrides local disabled variable.
A separate fix could be prepared for this into `7.7`